primitives/trie/src/storage_proof.rs

```diff
@@ -32,27 +32,27 @@ use crate::LayoutV1 as Layout;
 /// the serialized nodes and performing the key lookups.
 #[derive(Debug, PartialEq, Eq, Clone, Encode, Decode, TypeInfo)]
 pub struct StorageProof {
-	trie_nodes: Vec<Vec<u8>>,
+	trie_nodes: BTreeSet<Vec<u8>>,
 }
 
 /// Storage proof in compact form.
 #[derive(Debug, PartialEq, Eq, Clone, Encode, Decode, TypeInfo)]
 pub struct CompactProof {
-	pub encoded_nodes: Vec<Vec<u8>>,
+	pub encoded_nodes: BTreeSet<Vec<u8>>,
```

Comment:
In compact proof order of encoded node is very important, changing it breaks the proof.

Reply:
It defines the structure of the trie, and should be deterministic (if not there is a bug)